### DISCUSSION TITLE
Is sign-up / sign-in possible in the following scenario?

### DISCUSSION THREAD

**Discussion by javac9** [opencrx/opencrx-documentation#20] [Q&A]
Suppose we set up a website (some JS framework + OpenCrx REST) and we would like to have many people sign up and then sign in. After someone signs up, they should be stored as `Contact` in the database. After that, maybe enable them to create their Company profile, etc.

I'm not sure if the above is possible because so far I've seen that `Contact` (or `Legal entity`, etc.) creation only happens from within the (official) OpenCrx app.

How would one go about achieving this?

**Comment by fschon** [comment]
Hi Javac9. I have some code that can get you started. How can i reach you personally? We might have some synergies in our projects. Lets chat. Franz

**Comment by javac9** [comment]
@wfro64 Sorry for the mention but please share your 2 cents on this one - at least a sentence. If impossible this could be a deal-breaker for what I need to do with OpenCrx. Thanks in advance!

**Comment by wfro64** [comment]
This scenario is of course possible since every step can be performed with a REST API call (see https://github.com/opencrx/opencrx-documentation/blob/master/Sdk/Rest.md):

* Create a new segment (for the new company): http://www.opencrx.org/opencrx/4.3/uml/opencrx-core/admin1/tn/010_Main.png.html
* Create a contact: http://www.opencrx.org/opencrx/4.3/uml/opencrx-core/account1/tn/010_Main.png.html and (email) addresses http://www.opencrx.org/opencrx/4.3/uml/opencrx-core/account1/tn/030_Addresses.png.html
* Create new user home for this contact: http://www.opencrx.org/opencrx/4.3/uml/opencrx-core/home1/tn/999_Segment.png.html for the newly created contact and set the initial password
* It is also possible to send the user a reset password token to their registered email address so that they can set their password (http://www.opencrx.org/opencrx/4.3/uml/opencrx-core/home1/tn/010_Main.png.html)

**Comment by javac9** [comment]
@wfro64  But what if I wanted to have all these new people register and login within the existing (for example: Standard) segment? 

Something like this:
- user visits register page, fills data in and submits the form
- a new Opencrx user should be created, along with a new `Contact` for this user
- then, the user should be able to log in with their login details
- after that, that user should be able to create his Company

Now, let's say one of the users would like to add another user (at that moment unrelated to them) as, for example, customer of his - all within the same Standard segment. Possible?

What might be consequences of having thousands such users within a single Segment?

**Comment by javac9** [comment]
> Hi Javac9. I have some code that can get you started. How can i reach you personally? We might have some synergies in our projects. Lets chat. Franz

Hi again, please contact me at **####@gmail.com**
Thanks, I hope as well to find synergy in our project, or at least exchange some experiences with OpenCrx!
